### PR TITLE
Namespace related bugfixes

### PR DESCRIPTION
**Pull request by Kidswiss**
## Summary

* This ensures that netpols created by `CreateNetworkPolicy()` will always create the networkpolicy to the sli namespace
* `.status.instanceNamespace` is now always populated via the `BootstrapInstanceNs()` function

## Checklist

- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog
- [x] Update tests.
- [x] Link this PR to related issues.

### CODE REVIEW

File: apis/vshn/v1/dbaas_vshn_redis.go [hunk 1]
```diff
@@ -193,6 +193,10 @@ func (redis *VSHNRedis) GetInstanceNamespace() string {
 	// GetMaintenanceDayOfWeek returns the currently set day of week
 }
 
+func (redis *VSHNRedis) SetInstanceNamespaceStatus() {
+	redis.Status.InstanceNamespace = redis.GetInstanceNamespace()
+}
+
```
Comment (wejdross) [inline apis/vshn/v1/dbaas_vshn_redis.go:199]: maybe:
func (v *VSHNRedis) SetInstanceNamespaceStatus() {
	v.Status.InstanceNamespace = v.GetInstanceNamespace()
}

to keep it the same for all services?

File: pkg/comp-functions/functions/vshnkeycloak/deploy.go [hunk 2]
```diff
@@ -215,7 +209,7 @@ func addPostgreSQL(svc *runtime.ServiceRuntime, comp *vshnv1.VSHNKeycloak) error
 		},
 	}
 
-	err = common.CreateNetworkPolicy([]string{comp.GetInstanceNamespace()}, pg.GetInstanceNamespace(), pg.GetName()+"-keycloak", svc)
+	err = common.CreateNetworkPolicy([]string{comp.GetInstanceNamespace()}, pg.GetInstanceNamespace(), comp.GetName(), svc)
```
Comment (Kidswiss) [inline pkg/comp-functions/functions/vshnkeycloak/deploy.go:212]: ah damn, this is unintentional...

**Review by wejdross — COMMENTED** [review]
Please test it as we use instanceNamespace in our documentatinon, otherwise looks good

File: apis/vshn/v1/dbaas_vshn_redis.go (hunk 1 above)
Comment (Kidswiss) [reply apis/vshn/v1/dbaas_vshn_redis.go:199]: The functions right below and above this one both use `redis *VSHNRedis`

Comment (wejdross) [reply apis/vshn/v1/dbaas_vshn_redis.go:199]: fine for me ^^

**Comment by Kidswiss** [comment]
@wejdross I've now adjusted all the receivers for all vshn types.

**Review by wejdross — APPROVED** [review]
I like it!

**Pull request merged**